app/backend/src/couchers/jobs/handlers.py

```diff
@@ -130,6 +137,61 @@ def process_generate_message_notifications(payload):
             )
 
 
+def process_event_creation_emails(payload):
+    """
+    Sends out emails to all users subscribed to a newly created event's cluster.
+    """
+    logger.info(f"Sending out emails for event_id = {payload.event_id}")
+
+    # don't send emails to global and regional events
+    excluded_cluster_parent_node_ids = {
+        "Global Community": 1,
+        "Americas": 4,
+        "Asia-Pacific": 7,
+        "Europe, Africa, and Middle East": 10,
+    }
+    # don't send emails to users more than max_radius away from an event if there are more than max_users subscribed
+    max_users = 1000
+    max_radius = 20000
+
+    with session_scope() as session:
+        event, cluster = session.execute(
+            select(Event, Cluster)
+            .join(ClusterEventAssociation, Event.id == ClusterEventAssociation.event_id)
```

Comment:
`ClusterEventAssociation` lets you have the same event related to many clusters (though that's not implemented right now), so this query should accommodate for that (or at least handle the case with a meaningful exception).

Reply:
I think since it's not implemented like this already and this is only a stop-gap solution for event notifications before we create a proper notifications framework (with frontend controls users can tweak their notification settings with), I think it's probably sufficient to leave the `.one()` filter, no? If there are multiple matches because what you're suggesting is implemented before notifications, in my mind, we want to at least send it to a cluster instead of throwing an exception. Maybe I could loop through each cluster, apply all of this logic to each, and then only grab unique users so that they aren't notified twice? Not sure how to apply the `max_users` constraint in this situation either.